single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultDiagramLabelProvider.java

```diff
@@ -195,28 +196,35 @@ private NodeDecorator getBranchStatusDecorator(Node node, Direction direction, S
     }
 
     private List<FeederInfo> buildFeederInfos(ThreeWindingsTransformer transformer, ThreeWindingsTransformer.Side side) {
-        return Arrays.asList(
-                new DirectionalFeederInfo(ARROW_ACTIVE, transformer.getTerminal(side).getP(), valueFormatter::formatPower),
-                new DirectionalFeederInfo(ARROW_REACTIVE, transformer.getTerminal(side).getQ(), valueFormatter::formatPower));
+        return this.buildFeederInfos(transformer.getTerminal(side));
     }
 
     private List<FeederInfo> buildFeederInfos(Injection<?> injection) {
-        return Arrays.asList(
-                new DirectionalFeederInfo(ARROW_ACTIVE, injection.getTerminal().getP(), valueFormatter::formatPower),
-                new DirectionalFeederInfo(ARROW_REACTIVE, injection.getTerminal().getQ(), valueFormatter::formatPower));
+        return this.buildFeederInfos(injection.getTerminal());
     }
 
     private List<FeederInfo> buildFeederInfos(Branch<?> branch, Branch.Side side) {
-        return Arrays.asList(
-                new DirectionalFeederInfo(ARROW_ACTIVE, branch.getTerminal(side).getP(), valueFormatter::formatPower),
-                new DirectionalFeederInfo(ARROW_REACTIVE, branch.getTerminal(side).getQ(), valueFormatter::formatPower));
+        return this.buildFeederInfos(branch.getTerminal(side));
     }
 
     private List<FeederInfo> buildFeederInfos(HvdcLine hvdcLine, NodeSide side) {
         HvdcConverterStation<?> hvdcConverterStation = side == NodeSide.ONE ? hvdcLine.getConverterStation1()
                                                                                         : hvdcLine.getConverterStation2();
+        return this.buildFeederInfos(hvdcConverterStation.getTerminal());
+    }
+
+    private List<FeederInfo> buildFeederInfos(Terminal terminal) {
+        if (this.layoutParameters.isDisplayCurrentFeederInfo()) {
+            return Arrays.asList(
```

Comment:
we rather use `List.of`

Reply:
If we use `List.of` here, we will have a problem later in `DefaultDiagramLabelProvider` when trying, under certain conditions, to reverse the list. 
Before this PR, `Arrays.asList` was already used here.
I used an `ArrayList `to easily factorize the code. Let me know if it suits you!